mesh/mesh_operators.cpp

```diff
@@ -157,6 +157,142 @@ int ThresholdDerefiner::ApplyImpl(Mesh &mesh)
 }
 
 
+int CoefficientRefiner::ApplyImpl(Mesh &mesh)
+{
+   int max_it = 1;
+   return PreprocessMesh(mesh, max_it);
+}
+
+int CoefficientRefiner::PreprocessMesh(Mesh &mesh, int max_it)
+{
+   int rank = 0;
+   MFEM_VERIFY(max_it > 0, "max_it must be strictly positive")
+   MFEM_VERIFY(coeff, "Coefficient is not set for CoefficientRefiner object")
+
+   int dim = mesh.Dimension();
+   L2_FECollection l2fec(order, dim);
+   FiniteElementSpace* l2fes = NULL;
+
+   bool par = false;
+
+#ifdef MFEM_USE_MPI
+   ParMesh* pmesh = dynamic_cast<ParMesh*>(&mesh);
+   if (pmesh && pmesh->Nonconforming())
+   {
+      par = true;
+      l2fes = new ParFiniteElementSpace(pmesh, &l2fec);
+      gf = new ParGridFunction(static_cast<ParFiniteElementSpace*>(l2fes));
```

Comment:
Since the `[Par]GridFunction` will be deleted before exiting this function perhaps it should only be declared locally rather than being member data.